server.js

```diff
@@ -1,43 +1,21 @@
-import bodyParser from 'body-parser';
-import community from './server/routes/community';
+import app from './app';
 import dotenv from 'dotenv';
-import express from 'express';
-import gitLabRoute from './server/helpers/gitlabRoute';
-import mongoose from 'mongoose';
-import passportRoute from './server/routes/passport';
-import path from 'path';
-import user from './server/routes/user';
-
-// ALLOW SIGNUP W/O CERT(S):
-export const isAllowedForDev = false;
-// temporarily set to true to
-// allow anyone in (for dev)
 
 dotenv.config();
 
-// initialize mongoDB
-mongoose.Promise = require('bluebird');
-
-mongoose.connect(process.env.MONGO_URL).then(
-  (res) => { console.log('Mongoose connected'); },
-  (err) => { console.error('Error connecting to MongoDB. Make sure MongoDB is running.'); }
-);
-
-// initialize Express app and setup routes
-const app = express();
-app.use(bodyParser.json());
-
-app.use(passportRoute);
-app.use(user);
-app.use(community);
-app.use(gitLabRoute);
-
-// server main app and statis assets:
-app.use(express.static(path.join(__dirname, '/client/build')));
-app.get('*', (req, res) => {
-  res.sendFile(path.join(__dirname, '/client/build/index.html'));
-});
+/*
+\\*
+    We separate our app and server files
+    so that supertest/jest won't listed
+    to the port after testing... \_o_/
+    reference: https://goo.gl/8G5tBH
```

Comment:
This does not actually seem to be working. Jest should disconnect from the server and exit after the tests run, but it doesn't &mdash; you have to <kbd>Ctrl</kbd> + <kbd>C</kbd> to exit. 

However, I had to change the setup from what is shown in [this tutorial](http://www.albertgao.xyz/2017/05/24/how-to-test-expressjs-with-jest-and-supertest/) slightly. There, he exports `app` from `app.js`, however to get this to work I had to export the server after `app.listen(...)` is called. Otherwise, I got a `TypeError: app.address is not a function` from the `testSession` (or previously `request` when I was using just plain `supertest`) call, AKA the test call to the route. 

Not sure exactly what the issue is here, but `.address()` is only a function on a listening server, so exporting `app` before it is listening should never work?